src/rpc.rs

```diff
@@ -0,0 +1,489 @@
+use crate::backend::Backend;
+use crate::common::{DHTEntity, CommonKeypair};
+use crate::group::Group;
+use crate::repo::Repo;
+
+use std::convert::TryInto;
+use std::fs;
+use tonic::async_trait;
+use tonic::transport::Server;
+use tonic::{Request, Response, Status};
+use tokio::sync::broadcast::error::RecvError;
+use futures::StreamExt;
+use tokio::fs::File;
+use tokio::io::{self, AsyncBufReadExt, BufReader};
+use anyhow::Result;
+use tracing::{error, info};
+use veilid_core::{
+    CryptoKey, CryptoSystem, CryptoSystemVLD0, CRYPTO_KEY_LENGTH, DHTRecordDescriptor, RoutingContext, SharedSecret, VeilidAPI, VeilidUpdate, VeilidAppCall, VeilidState,
+};
+use anyhow::anyhow;
+use base64::engine::general_purpose::URL_SAFE_NO_PAD;
+use base64::Engine as _;
+use prost::Message;
+use iroh_blobs::Hash;
+
+use tonic_reflection::server::Builder;
+
+tonic::include_proto!("rpc");
+pub const FILE_DESCRIPTOR_SET: &[u8] = include_bytes!("../descriptor.bin");
+
+#[derive(Clone)]
+pub struct RpcService {
+    backend: Backend,
+    keypair: CommonKeypair,
+    routing_context: RoutingContext,
+    crypto_system: CryptoSystemVLD0,
+    dht_record: DHTRecordDescriptor,
+}
+
+impl RpcService {
+    async fn get_group(&self) -> Result<Group> {
```

Comment:
RPC service should expose all the groups in the backend over RPC

Reply:
yes, instead of getting a single group, lets get them all

Reply:
Does this `list_groups` [implementation](https://github.com/OpenArchive/save-dweb-backend/blob/create-backup-server/src/rpc.rs#L335-L348) meet the requirements for exposing all backend groups over RPC? 

It currently retrieves and returns all group IDs.

Reply:
Yes that's perfect